### PR TITLE
PRM Automated Tests - LPS-179357 & LPS-180526 |master

### PR DESCRIPTION
**Pull request by marinaasant0s**
[LPS-179357](https://issues.liferay.com/browse/LPS-179357) - ClaimApprovedValueAppearInDashboard
[LPS-180526](https://issues.liferay.com/browse/LPS-180526) - MDFClaimedValueAppearInDashboard

### CODE REVIEW

**Comment by marinaasant0s** [comment]
ci:test:sf

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-partner-portal/osb-site-initializer-partner-portal-test/src/testFunctional/macros/PRMMDFClaim.macro [hunk 1]
```diff
@@ -230,6 +230,17 @@ definition {
...
+	macro getClaimRequestID {
+		var claimId = selenium.getText("PRMMDFClaim#CLAIM_REQUEST_ID");
+
+		var claimId = StringUtil.extractDigits(${claimId});
+
+		return ${claimId};
+
+		echo("Claim: ${claimId}");
+	}
+
```
Comment (AfonsoHSantos) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-partner-portal/osb-site-initializer-partner-portal-test/src/testFunctional/macros/PRMMDFClaim.macro:243]: Hi @marinaasant0s, the path `PRMMDFClaim.getClaimIdOnDetailPage` already do it

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-partner-portal/osb-site-initializer-partner-portal-test/src/testFunctional/macros/PRMDashboards.macro [hunk 2]
```diff
@@ -12,11 +12,18 @@ definition {
...
 		return ${legendValue};
 	}
 
+	@summary = "Get the value of the legend in Dashboards"
+	macro scrollViewDashboard {
+		ScrollWebElementIntoView(
+			key_dashboardmdf = ${viewDashboard},
+			locator1 = "PRMDashboards#SCROLL_VIEW_DASHBOARD");
+	}
+
```
Comment (AfonsoHSantos) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-partner-portal/osb-site-initializer-partner-portal-test/src/testFunctional/macros/PRMDashboards.macro:28]: Hi @marinaasant0s, the scroll should be added inside the macro `PRMDashboards.getLegendValue`. Using the path `PRMDashboards#LEGEND_VALUE`

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-partner-portal/osb-site-initializer-partner-portal-test/src/testFunctional/tests/PRMDashboards.testcase [hunk 3]
```diff
@@ -99,6 +99,78 @@ definition {
 		}
 	}
 
+	@description = "This is a test for LPS-180526. Verify that the value of Claimed MDFs are displayed on MDF Dashboard"
+	@priority = 4
+	test MDFClaimedValueAppearInDashboard {
+		var budgetValue = 2000;
+
+		task ("Given the Channel Marketing Manager get the current amount of Claim MDF Value") {
```
Comment (AfonsoHSantos) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-partner-portal/osb-site-initializer-partner-portal-test/src/testFunctional/tests/PRMDashboards.testcase:207]: Hi @marinaasant0s, please do this action with the PMU user

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-partner-portal/osb-site-initializer-partner-portal-test/src/testFunctional/tests/PRMDashboards.testcase [hunk 4]
```diff
@@ -99,6 +99,78 @@ definition {
...
+			PRMNavigator.openSitePage(pageName = "Home");
+
+			PRMDashboards.scrollViewDashboard(viewDashboard = "Market Development Funds");
+
+			var initialValue = PRMDashboards.getLegendValue(
+				chartName = "Request",
+				legendName = "Claimed");
+		}
+
+		task ("And a new MDF Request created and approved it") {
```
Comment (AfonsoHSantos) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-partner-portal/osb-site-initializer-partner-portal-test/src/testFunctional/tests/PRMDashboards.testcase:217]: Hi @marinaasant0s, please rename this task to `And a new MDF Request is created and approved`

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-partner-portal/osb-site-initializer-partner-portal-test/src/testFunctional/tests/PRMDashboards.testcase [hunk 5]
```diff
@@ -99,6 +99,78 @@ definition {
...
+
+			PRMMDFRequest.goToMDF(campaignName = "Campaign Name 1");
+		}
+
+		task ("And the User create a new MDF Claim") {
+			PRMMDFClaim.createNewClaim(
+				listLeads = "Document_1.xlsx",
+				reimbursementInvoice = "Document_1.png",
+				thirdInvoice = "Document_1.png");
+		}
```
Comment (AfonsoHSantos) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-partner-portal/osb-site-initializer-partner-portal-test/src/testFunctional/tests/PRMDashboards.testcase:247]: Hi @marinaasant0s, please add all these steps inside the same task `And the PMU creates a new claim in the submitted MDF`

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-partner-portal/osb-site-initializer-partner-portal-test/src/testFunctional/tests/PRMDashboards.testcase [hunk 6]
```diff
@@ -52,6 +52,106 @@ definition {
...
+			}
+		}
+
+		task ("Then the actual amount of approved MDF Claim value was increased") {
+			TestUtils.assertEquals(
+				actual = ${newValue},
+				expected = ${expectedValue});
+		}
+	}
+
```
Comment (AfonsoHSantos) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-partner-portal/osb-site-initializer-partner-portal-test/src/testFunctional/tests/PRMDashboards.testcase:154]: Hi @marinaasant0s, please apply the request changes from the previous commit here.

**Review by AfonsoHSantos — CHANGES_REQUESTED** [review]

**Comment by marinaasant0s** [comment]
ci:test:sf

**Review by AfonsoHSantos — APPROVED** [review]
Hi @marinaasant0s, thanks for working on it!
The code looks good to me, so I'm forwarding it.

giphy

**Pull request closed without merging**